modules/programs/kitty.nix

```diff
@@ -111,48 +134,87 @@ in {
+          <link xlink:href="https://sw.kovidgoyal.net/kitty/shell-integration"/>
+          for more details.
+        '';
+        example = "no-cursor";
+      };
+      enableBashIntegration = mkEnableOption "Kitty bash integration" // {
+        default = true;
+      };
+      enableFishIntegration = mkEnableOption "Kitty fish integration" // {
+        default = true;
+      };
+      enableZshIntegration = mkEnableOption "Kitty zsh integration" // {
+        default = true;
+      };
+    };
+
     extraConfig = mkOption {
       default = "";
       type = types.lines;
       description = "Additional configuration to add.";
     };
   };
 
-  config = mkIf cfg.enable {
-    home.packages = [ cfg.package ] ++ optionalPackage cfg.font;
-
-    xdg.configFile."kitty/kitty.conf" = {
-      text = ''
-        # Generated by Home Manager.
-        # See https://sw.kovidgoyal.net/kitty/conf.html
-      '' + concatStringsSep "\n" ([
-
-        (optionalString (cfg.font != null) ''
-          font_family ${cfg.font.name}
-          ${optionalString (cfg.font.size != null)
-          "font_size ${toString cfg.font.size}"}
-        '')
-
-        (optionalString (cfg.theme != null) ''
-          include ${pkgs.kitty-themes}/${
-            (head (filter (x: x.name == cfg.theme) (builtins.fromJSON
-              (builtins.readFile "${pkgs.kitty-themes}/themes.json")))).file
-          }
-        '')
-        (toKittyConfig cfg.settings)
-        (toKittyKeybindings cfg.keybindings)
-        (toKittyEnv cfg.environment)
-        cfg.extraConfig
-      ]);
-    } // optionalAttrs pkgs.stdenv.hostPlatform.isLinux {
-      onChange = ''
-        ${pkgs.procps}/bin/pkill -USR1 -u $USER kitty || true
-      '';
-    };
-
-    xdg.configFile."kitty/macos-launch-services-cmdline" = mkIf
-      (cfg.darwinLaunchOptions != null && pkgs.stdenv.hostPlatform.isDarwin) {
-        text = concatStringsSep " " cfg.darwinLaunchOptions;
+  config = mkIf cfg.enable (mkMerge [
+    {
+      home.packages = [ cfg.package ] ++ optionalPackage cfg.font;
+
+      xdg.configFile."kitty/kitty.conf" = {
+        text = ''
+          # Generated by Home Manager.
+          # See https://sw.kovidgoyal.net/kitty/conf.html
+        '' + concatStringsSep "\n" ([
+
+          (optionalString (cfg.font != null) ''
+            font_family ${cfg.font.name}
+            ${optionalString (cfg.font.size != null)
+            "font_size ${toString cfg.font.size}"}
+          '')
+
+          (optionalString (cfg.theme != null) ''
+            include ${pkgs.kitty-themes}/${
+              (head (filter (x: x.name == cfg.theme) (builtins.fromJSON
+                (builtins.readFile "${pkgs.kitty-themes}/themes.json")))).file
+            }
+          '')
+
+          (optionalString (cfg.shellIntegration.mode != "disabled") ''
+            # Shell integration is sourced and configured manually
+            shell_integration disabled
+          '')
```

Comment:
If I understand the manual correctly, we could set that to `no-rc ${shellIntegration.mode}` so that `KITTY_SHELL_INTEGRATION` is set to the correct value automatically?